### ISSUE TITLE
Unable to set colors above index 15 with OSC 4, take 2

### ISSUE THREAD

**Issue by egmontkob** [microsoft/terminal#3717]
# Environment

```none
Windows build number: 10.0.18362.0
Windows Terminal version (if applicable): 0.7.3291.0
```

# Steps to reproduce

Allegedly already fixed in #313, but doesn't work in v0.7:

    echo -ne '\e]4;15;rgb:ab/cd/ef\e\\'
    echo -ne '\e]4;16;rgb:ab/cd/ef\e\\'

# Expected behavior

Set palette index 15 and 16 to the given color.

# Actual behavior

Color index 15 is set to the given value (works for 0..15), index 16 remains unchanged (doesn't work for 16..255).

**Comment by zadjii-msft** [comment]
My guess is that setting the colors >15 probably works in conhost, but doesn't roundtrip through conpty.

**Comment by j4james** [comment]
This doesn't really work on conhost either. You can technically set all the colors, and any future use of those colors will be displayed correctly, but your existing output won't update. In fact even the first 16 colors won't update if you've set them using the 256-color index syntax. I think that is what issue #1223 is all about.

Fixing #1223 is a bit of a nightmare, but I put together a hack solution just to see what would happen, and it also partially resolves the issue in the Windows Terminal. Sometimes, though, the colors just don't refresh, and I don't know enough about that side of the code to figure out why.

**Comment by DHowett-MSFT** [comment]
I wonder: does fixing #2661 and then triggering a full invalidate on all screenful renderers help #1223? It seems like it may..

**Comment by j4james** [comment]
Actually I was going to say fixing #1223 is probably a prerequisite for fixing #2661. If you want to preserve the color index when passing the attributes through conpty, that assumedly requires that the index was already preserved on the conhost side when writing to the output buffer - and that's essentially the fix for #1223.
